### PR TITLE
CLN: Remove unnecessary uses of pd. in tests

### PR DESCRIPTION
**Pull request by jschendel**
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

- Mostly edits along the lines of `pd.Series` -> `Series` for things that were already imported and had redundant uses of `pd`.  
- I cleaned some of the `import` statements, but I didn't import anything new, though that could be done in the future for further cleaning. 
- Was able to shorten some multi-line statements for a net deletion of ~100 lines without changing the content of any tests.
- Only did the base level `tests` and `tests/indexing` so far.

### CODE REVIEW

File: pandas/tests/test_sorting.py [hunk 1]
```diff
@@ -418,7 +415,7 @@ def test_mixed_integer_from_list(self):
     def test_unsortable(self):
         # GH 13714
         arr = np.array([1, 2, datetime.now(), 0, 3], dtype=object)
-        if compat.PY2 and not pd._np_version_under1p10:
+        if compat.PY2 and not compat.numpy._np_version_under1p10:
```
Comment (jreback) [inline pandas/tests/test_sorting.py]: I would import _np_version.... directly up top

**Review by jreback — CHANGES_REQUESTED** [review]
happy to take this (one small comment). Ideally we could add a linting test for these, something along the lines of (of course have to augment with things we check for, which would things like DataFrame, Index, etc.).

you could start off but explicity using just certain files / dirs for now (IOW as you do them).

you can put this after the ``np.testing`` checks
``grep -r -E --include '*.py' 'pd\.Series' pandas/tests/``

**Comment by jreback** [comment]
@jschendel just that one change is fine for now. the lint checks can be in a followup (or here if you can/want).

**Comment by jschendel** [comment]
@jreback : made the one change. not familiar with the lint checks, so I'll hold off until a followup.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jschendel would love to come back with something that we can add to ``ci/lint.sh`` to check the codebase automatically.

File: pandas/tests/test_categorical.py [hunk 2]
```diff
@@ -1013,7 +1009,7 @@ def test_tab_complete_warning(self, ip):
         pytest.importorskip('IPython', minversion="6.0.0")
         from IPython.core.completer import provisionalcompleter
 
-        code = "import pandas as pd; c = pd.Categorical([])"
+        code = "import pandas as pd; c = Categorical([])"
```
Comment (jbrockmendel) [inline pandas/tests/test_categorical.py:1012]: @jschendel @jreback why is removing pd OK here?  We have a couple tests with this pattern that look fishy to me, but they are passing on master so im confused

**Pull request merged**